src/v/cluster/rm_stm.cc

```diff
@@ -1914,6 +1920,41 @@ ss::future<model::record_batch> rm_stm::make_checkpoint_batch() {
     co_return std::move(builder).build();
 }
 
+ss::future<model::record_batch>
+rm_stm::make_checkpoint_applied_batch(model::term_id term) {
+    iobuf key;
+    co_await serde::write_async(key, term);
+    storage::record_batch_builder builder(
+      model::record_batch_type::tx_checkpoint_applied, model::offset(0));
+    builder.set_control_type();
+    builder.add_raw_kv(std::move(key), iobuf{});
+    co_return std::move(builder).build();
+}
+
+ss::future<> rm_stm::replicate_checkpoint_applied(model::term_id term) {
+    if (!_parked_checkpointed_mem_state) {
```

Comment:
let's move the check a level up or rename `replicate_checkpoint_applied` to `maybe_replicate_checkpoint_applied`